rotkehlchen/data/importer.py

```diff
@@ -1275,3 +1323,99 @@ def import_uphold_transactions_csv(self, filepath: Path, **kwargs: Any) -> Tuple
+        (@yabirgb) this file can only contain confirmed trades.
+        - UnknownAsset
+        - DeserializationError
+        """
+        if csv_row['Status'] == 'Canceled':
+            return
+        formatstr = kwargs.get('timestamp_format')
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date/Time'],
+            formatstr=formatstr if formatstr is not None else '%d %b %Y %H:%M:%S',
+            location='Bisq',
+        )
+        # Get assets and amounts sold
+        offer = csv_row['Offer type'].split()
+        assets1_symbol, assets2_symbol = csv_row['Market'].split('/')
+        if offer[0] == 'Sell':
+            trade_type = TradeType.SELL
+            if offer[1] == assets1_symbol:
+                base_asset = symbol_to_asset_or_token(assets1_symbol)
+                quote_asset = symbol_to_asset_or_token(assets2_symbol)
+            else:
+                base_asset = symbol_to_asset_or_token(assets2_symbol)
+                quote_asset = symbol_to_asset_or_token(assets1_symbol)
+
+            if base_asset == A_BTC:
+                buy_amount = deserialize_asset_amount(csv_row['Amount'])
+            else:
+                buy_amount = deserialize_asset_amount(csv_row['Amount in BTC'])
+
+        else:
+            trade_type = TradeType.BUY
+            if offer[1] == assets1_symbol:
+                base_asset = symbol_to_asset_or_token(assets1_symbol)
+                quote_asset = symbol_to_asset_or_token(assets2_symbol)
+            else:
+                base_asset = symbol_to_asset_or_token(assets2_symbol)
+                quote_asset = symbol_to_asset_or_token(assets1_symbol)
+
+            if base_asset == A_BTC:
+                buy_amount = deserialize_asset_amount(csv_row['Amount in BTC'])
+            else:
+                buy_amount = deserialize_asset_amount(csv_row['Amount'])
+
+        rate = Price(deserialize_asset_amount(csv_row['Price']))
+        # Get trade fee
+        if len(csv_row['Trade Fee BSQ']) != 0:
+            fee_amount = deserialize_fee(csv_row['Trade Fee BSQ'])
+            fee_currency = A_BSQ
+        else:
+            fee_amount = deserialize_fee(csv_row['Trade Fee BTC'])
+            fee_currency = A_BTC
+
+        trade = Trade(
+            timestamp=timestamp,
+            location=Location.BISQ,
+            base_asset=base_asset,
+            quote_asset=quote_asset,
+            trade_type=trade_type,
+            amount=buy_amount,
+            rate=rate,
+            fee=fee_amount,
+            fee_currency=fee_currency,
+            link='',
+            notes=f'ID: {csv_row["Trade ID"]}',
+        )
+        self.db.add_trades([trade])
+
+    def import_bisq_trades_csv(self, filepath: Path, **kwargs: Any) -> Tuple[bool, str]:
+        """
+        Information for the values that the columns can have has been obtained from
+        the issue in github #1674
+        """
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
```

Comment:
Question: Do you remember why we open all the csvs in this file with `'utf-8-sig'` encoding?

Reply:
I was here after this trend but I believe is related to the comment here https://docs.python.org/3/library/codecs.html#encodings-and-unicode (search for utf-8-sig) about windows special utf-8